### PR TITLE
[iOS] Change API endpoint according to build configuration

### LINKED ISSUE #674: Use the production API endpoint for iOS app as well
## Kind (Required)

- New feature

 ## Overview (Required)

- Change api endpoints based on Schema for iOS app
- https://github.com/DroidKaigi/conference-app-2019/blob/master/data/api-impl/build.gradle#L73 should be used for the release build

 ## Links	

- https://github.com/DroidKaigi/conference-app-2019/blob/master/data/api-impl/src/iosMain/kotlin/io/github/droidkaigi/confsched2019/data/api/ApiEndpoint.kt#L3

### PR DESCRIPTION
**Pull request by hironytic**
## Issue
- close #674 

## Overview (Required)
- In addition to `src/iOSMain/kotlin`, one of
  `src/iOSMain/kotlinDebug`
   or
  `src/iOSMain/kotlinRelease`
  is also source directory now, according to Xcode configuration (Debug or Release)
- Move`ApiEndpoint.kt` to `kotlinDebug` and `kotlinRelease` , making it return appropriate endpoint

I couldn't find out the way to tell the configuration value to Kotlin source code. So I decided to switch  the whole source file itself.
`project.findProperty("XCODE_CONFIGURATION")` is used for detecting Xcode configuration. I don't know much about it 😇 but it is already used in `packForXCode` task in ioscombined/build.gradle.

Though I've also changed the api-impl/build.gradle, I think is not used for iOS build 🤔

### CODE REVIEW

File: data/api-impl/src/iosMain/kotlinDebug/io/github/droidkaigi/confsched2019/data/api/ApiEndpoint.kt [hunk 1]
```diff
@@ -1,5 +1,3 @@
 package io.github.droidkaigi.confsched2019.data.api
 
-// TODO: Replace with code getting endpoint from iOS application build configurations.
```
Comment (jmatsu) [inline data/api-impl/src/iosMain/kotlinDebug/io/github/droidkaigi/confsched2019/data/api/ApiEndpoint.kt]: 👍

File: frontend/ioscombined/build.gradle [hunk 2]
```diff
@@ -53,8 +53,14 @@ kotlin {
             }
         }
         iOSMain {
+            final String mode = project.findProperty("XCODE_CONFIGURATION")?.toUpperCase() ?: 'DEBUG'
             projectsList.forEach {
                 kotlin.srcDirs += "${project(it).projectDir}/src/iosMain/kotlin"
+                if (mode == "RELEASE") {
```
Comment (jmatsu) [inline frontend/ioscombined/build.gradle]: Sounds smart. 🆒

File: data/api-impl/build.gradle [hunk 3]
```diff
@@ -50,6 +50,13 @@ kotlin {
             }
         }
         iOSMain {
+            final String mode = project.findProperty("XCODE_CONFIGURATION")?.toUpperCase() ?: 'DEBUG'
```
Comment (jmatsu) [inline data/api-impl/build.gradle]: `projectsList` in ioscombined can handle this, so I guess this diff is not required. Didn't it work without this chunk? 👀

Comment (hironytic) [reply data/api-impl/build.gradle]: Yes. It works without this.

**Comment by hironytic** [comment]
> As you notied, could you please try removing the hunk of changes?

OK. Wait a minute.

**Pull request merged**